libraries/AP_RCProtocol/AP_RCProtocol.cpp

```diff
@@ -208,7 +208,9 @@ void AP_RCProtocol::check_added_uart(void)
     for (uint8_t i=0; i<n; i++) {
         int16_t b = added.uart->read();
         if (b >= 0) {
-            process_byte(uint8_t(b), added.baudrate);
+            // get an estimate of when the i-th byte was received
+            uint32_t timestamp_us = added.uart->receive_time_constraint_us(i+1);
```

Comment:
this should be:
  added.uart->receive_time_constraint_us(n-i)
as the first bytes arrived earlier than the later bytes

Reply:
Tridge my first try was n-i as you suggested but it did not detect any gaps

Reply:
better to call added.uart->receive_time_constraint_us(n) once before the loop, like this:
 uint32_t start_us = uint32_t(added.uart->receive_time_constraint_us(n));
 uint32_t us_per_byte = (10U * 1000000U) / added.baudrate;

then in loop do this:
  uint32_t byte_us = start_us + i * us_per_byte;

and pass byte_us to the process_byte() call